transport/server.cc

```diff
@@ -237,24 +239,44 @@ cql_server::do_accepts(int which, bool keepalive, socket_address server_addr) {
             fd.set_keepalive(keepalive);
             auto conn = make_shared<connection>(*this, server_addr, std::move(fd), std::move(addr));
             ++_connects;
-            ++_connections;
-            // Move connection to the background, monitor for lifetime and errors.
-            (void)conn->process().then_wrapped([this, conn] (future<> f) {
+            (void)advertise_new_connection(conn).then([this, conn] {
+                ++_connections;
+                // Monitor connection's lifetime and errors in the background.
+                return conn->process().then_wrapped([this, conn] (future<> f) {
+                    try {
+                        f.get();
+                    } catch (...) {
+                        clogger.debug("connection error: {}", std::current_exception());
+                    }
+                });
+            }).then([this, conn] {
                 --_connections;
-                try {
-                    f.get();
-                } catch (...) {
-                    clogger.debug("connection error: {}", std::current_exception());
-                }
+                return unadvertise_connection(conn);
```

Comment:
I think an exception here won't be logged, since the future is dropped (cast to void). So need to log it explictly here.

(there is a last ditch logger for discarded exceptional futures, but we try not to rely on it, since it doesn't provide any context to the user).

We may want to log and ignore errors when adding connections, too, since a problem in that table can kill the node by refusing any new connections, including those that attempt to fix the table.

Reply:
please see comment above (or, even better, the code)